pkg/collector/vg_collector.go

```diff
@@ -0,0 +1,145 @@
+			[]string{"name"}, nil,
+		),
+		vgSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "total_size_bytes"),
+			"LVM VG total size in bytes",
+			[]string{"name"}, nil,
+		),
+		vgLvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "lv_count"),
+			"Number of LVs in VG",
+			[]string{"name"}, nil,
+		),
+		vgPvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "pv_count"),
+			"Number of PVs in VG",
+			[]string{"name"}, nil,
+		),
+		vgMaxLvMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "max_lv_count"),
+			"LMaximum number of LVs allowed in VG or 0 if unlimited",
+			[]string{"name"}, nil,
+		),
+		vgMaxPvMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "max_pv_count"),
+			"Maximum number of PVs allowed in VG or 0 if unlimited",
+			[]string{"name"}, nil,
+		),
+		vgSnapCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "snap_count"),
+			"Number of snapshots in VG",
+			[]string{"name"}, nil,
+		),
+		vgMissingPvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "missing_pv_count"),
+			"Number of PVs in VG which are missing",
+			[]string{"name"}, nil,
+		),
+		vgMetadataCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_count"),
+			"Number of metadata areas on this VG",
+			[]string{"name"}, nil,
+		),
+		vgMetadataUsedCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_used_count"),
+			"Number of metadata areas in use on this VG",
+			[]string{"name"}, nil,
+		),
+		vgMetadataFreeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_free_size_bytes"),
+			"Free metadata area space for this VG in bytes",
+			[]string{"name"}, nil,
+		),
+		vgMetadataSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_total_size_bytes"),
+			"Size of smallest metadata area for this VG in bytes",
+			[]string{"name"}, nil,
+		),
+		vgPermissionsMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "permission"),
+			"VG permissions: [-1: undefined], [0: writeable], [1: read-only]",
+			[]string{"name"}, nil,
+		),
+		vgAllocationPolicyMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "allocation_policy"),
+			"VG allocation policy: [-1: undefined], [0: normal], [1: contiguous], [2: cling], [3: anywhere], [4: inherited]",
+			[]string{"name"}, nil,
+		),
+	}
+}
+
+func (c *vgCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.vgFreeMetric
+	ch <- c.vgSizeMetric
+	ch <- c.vgLvCountMetric
+	ch <- c.vgPvCountMetric
+	ch <- c.vgMaxLvMetric
+	ch <- c.vgMaxPvMetric
+	ch <- c.vgSnapCountMetric
+	ch <- c.vgMissingPvCountMetric
+	ch <- c.vgMetadataCountMetric
+	ch <- c.vgMetadataUsedCountMetric
+	ch <- c.vgMetadataFreeMetric
+	ch <- c.vgMetadataSizeMetric
+	ch <- c.vgPermissionsMetric
+	ch <- c.vgAllocationPolicyMetric
+}
+
+func (c *vgCollector) Collect(ch chan<- prometheus.Metric) {
+	vgList, err := lvm.ListLVMVolumeGroup(false)
```

Comment:
false means?

Reply:
This is to prevent the refreshing of the cache in every function call. Since I guess this is being done regularly by the node plugin itself.